### PR TITLE
Crystal structure: single crystals, unit cells, and rotation sets

### PR DESCRIPTION
**Pull request by lukaspie**
This PR takes all of the scattered unit cell and single crystal descriptions in NXsample and creates new base classes: NXsingle_crystal (to be used if a sample (component) is a single crystal) and NXunit_cell.

NXsingle_crystal and NXunit_cell make use of proposals introduced in (PR 51)[https://github.com/FAIRmat-NFDI/pull/51]:

NXrotation_set as a detailed description of single crystal orientation
short-hand (numpy-like) description of dimensions
Instead of
dimensions:
rank=1
dim=[1,3],
we can write dim=(3,) in the yaml files now.

### CODE REVIEW

File: base_classes/nyaml/NXsample.yaml
Comment (mkuehbach) [inline base_classes/nyaml/NXsample.yaml:1]: I am confused here I thought NXsample is already part of another PR, is this just to assure that the sample is not removed again during the merge?

**Review by mkuehbach — CHANGES_REQUESTED** [review]

File: base_classes/nyaml/NXsample.yaml
Comment (lukaspie) [reply base_classes/nyaml/NXsample.yaml:1]: This adds NXsingle_crystal to NXsample, which is not part of the other PR. After merging both, it should all be in NXsample.

Comment (lukaspie) [reply base_classes/nyaml/NXsample.yaml:1]: I can just remove the YAML here, then it is possibly clearer.

**Review by mkuehbach — APPROVED** [review]
lgtm

**Pull request merged**